src/blocks/homepage-articles/templates/articles-loop.php

```diff
@@ -23,7 +30,8 @@ function( $data ) {
 			$post_counter++;
 			echo Newspack_Blocks::template_inc( __DIR__ . '/article.php', array( 'attributes' => $attributes ) ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
 		}
-		wp_reset_postdata();
+
+		$wp_query = $main_query; // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
```

Comment:
```suggestion
		$wp_query = $main_query; // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
		wp_reset_query();
```

I believe the query needs another reset after we put it back.

Reply:
> I believe the query needs another reset after we put it back.

Works fine when I do it 2x, but I'm not sure I completely understand why it's needed? By replacing the global with the stashed query aren't we destroying everything that just happened anyway?

Reply:
Good point. At minimum I think the `wp_reset_postdata` is still necessary, for resetting the global `$post`.

Reply:
Otherwise WP will think that we're still using the last post of the query, so we'll get weird side-effects in the footer and other areas under the homepage block.